### PR TITLE
report listener and storage types as found keys

### PR DESCRIPTION
**Pull request by HridoyRoy**
This PR fixes https://github.com/hashicorp/vault/issues/15331

### CODE REVIEW

File: command/server/config.go [hunk 1]
```diff
@@ -967,3 +968,8 @@ func (c *Config) Prune() {
 		c.Telemetry.UnusedKeys = nil
 	}
 }
+
+func (c *Config) found(s, k string) {
```
Comment (hghaf099) [inline command/server/config.go:972]: From both instances that used this function, wondering when would `s` and `k` be different?

**Review by hghaf099 — COMMENTED** [review]
I just have one question, otherwise looks good!

**Review by sgmiller — APPROVED** [review]
Approve apart from hghaf099's comment

File: command/server/config.go (hunk 1 above)
Comment (HridoyRoy) [reply command/server/config.go:972]: My primary reason for introducing this function is to keep parity with our helper in SharedConfig:

```func (c *SharedConfig) found(s, k string) {
	delete(c.UnusedKeys, s)
	c.FoundKeys = append(c.FoundKeys, k)
}```

There are often cases where we transform the capitalization of keys in configs. For example, `result.found("seal", "Seal")` and `result.found("entropy", "Entropy")`. While we currently aren't doing this for the top level config (only the SharedConfig) I'd like to keep this functionality for future development.

**Pull request merged**